src/app/pages/receive/receive-modal.tsx

```diff
@@ -0,0 +1,104 @@
+import { useCurrentAccountStxAddressState } from '@app/store/accounts/blockchain/stacks/stacks-account.hooks';
+
+import { ReceiveItem } from './components/receive-item';
+import { ReceiveItems } from './components/receive-items';
+
+type ReceiveModal = 'full' | 'collectible';
+
+interface ReceiveModalProps {
+  type?: 'full' | 'collectible';
+}
+
+export function ReceiveModal({ type = 'full' }: ReceiveModalProps) {
+  const analytics = useAnalytics();
+  const location = useLocation();
+  const navigate = useNavigate();
+  const btcAddressNativeSegwit = useCurrentAccountNativeSegwitAddressIndexZero();
+  const stxAddress = useCurrentAccountStxAddressState();
+  const accountIndex = get(location.state, 'accountIndex', undefined);
+  const btcAddressTaproot = useZeroIndexTaprootAddress(accountIndex);
+
+  const { onCopy: onCopyBtc } = useClipboard(btcAddressNativeSegwit);
+  const { onCopy: onCopyStx } = useClipboard(stxAddress);
+  const { onCopy: onCopyOrdinal } = useClipboard(btcAddressTaproot);
+
+  function copyToClipboard(copyHandler: () => void, tracker = 'copy_address_to_clipboard') {
+    void analytics.track(tracker);
+    toast.success('Copied to clipboard!');
+    copyHandler();
+  }
+
+  const title = type === 'full' ? 'Select asset to receive' : 'Add collectible';
+
+  return (
+    <BaseDrawer title={title} isShowing onClose={() => navigate('../')}>
+      <Box mx="extra-loose">
+        {type === 'full' && (
+          <ReceiveItems title="Tokens">
+            <ReceiveItem
+              address={btcAddressNativeSegwit}
+              icon={<BtcIcon />}
+              dataTestId={HomePageSelectors.ReceiveBtcNativeSegwitQrCodeBtn}
+              onCopyAddress={() => copyToClipboard(onCopyBtc)}
+              onClickQrCode={() => navigate(RouteUrls.ReceiveBtc)}
+              title="Bitcoin"
+            />
+            <ReceiveItem
+              address={stxAddress}
+              icon={<StxAvatar />}
+              dataTestId={HomePageSelectors.ReceiveStxQrCodeBtn}
+              onCopyAddress={() => copyToClipboard(onCopyStx)}
+              onClickQrCode={() => navigate(RouteUrls.ReceiveStx)}
+              title="Stacks"
+            />
+          </ReceiveItems>
+        )}
+        <ReceiveItems title={type === 'full' ? 'Collectibles' : undefined}>
+          <ReceiveItem
+            address={btcAddressTaproot}
+            icon={<OrdinalIcon />}
+            dataTestId={HomePageSelectors.ReceiveBtcTaprootQrCodeBtn}
+            onCopyAddress={() =>
+              copyToClipboard(onCopyOrdinal, 'select_stamp_to_add_new_collectible')
+            }
+            onClickQrCode={() => {
+              void analytics.track('select_inscription_to_add_new_collectible');
+              navigate(RouteUrls.ReceiveCollectibleOrdinal, { state: { btcAddressTaproot } });
+            }}
+            title="Ordinal inscription"
+          />
+          <ReceiveItem
```

Comment:
I don't pass `dataTestId`s to these two as I didn't add new tests for them. 

I think it's basically the same as the tests above for `HomePageSelectors.ReceiveBtcNativeSegwitQrCodeBtn` and `HomePageSelectors.ReceiveStxQrCodeBtn`. 

I can add some more tests if desired to make sure we catch each button scenario